src/MessageTrait.php

```diff
@@ -66,7 +66,7 @@ public function getProtocolVersion()
      * new protocol version.
      *
      * @param string $version HTTP protocol version
-     * @return self
+     * @return static
```

Comment:
Just joining the conversation: since the interface defines that a [`MessageInterface`](https://github.com/php-fig/http-message/blob/85d63699f0dbedb190bbd4b0d2b9dc707ea4c298/src/MessageInterface.php#L39) is returned, so should also the trait providing the concrete impl :-)